### PR TITLE
Filtering by Stewardship Organisation, search by collection and discussion

### PR DESCRIPTION
**Pull request by s-i-l-k-e**
Fixes issues: #

Sanity checks:
- [ ] Have tests been run locally?
- [ ] Does this change any javascript? If so, have you checked that the UI works as expected?
- [ ] If there are data model changes, are relevant data standards referenced - in code?
- [ ] If there are deviations from any specified standards, are the reasons for changes documented?
- [ ] If there are UI changes have you documented whats changed so the help documentation can be updated?

### CODE REVIEW

File: python/aristotle-metadata-registry/aristotle_mdr/forms/search.py [hunk 1]
```diff
@@ -158,14 +156,19 @@ def models(self, *mods):
...
     def apply_permission_checks(self, user=None, public_only=False, user_workgroups_only=False):
+        """"
+        Apply permission checks by altering the SearchQuery
+        """
         sqs = self
         q = SQ(is_public=True)
         q |= SQ(published_date_public__lte=timezone.now())
         if user is None or user.is_anonymous():
-            # Regular users can only see public items, so boot them off now.
+            # Regular users can only see public items, so filter only on the public items.
```
Comment (LegoStormtroopr) [inline python/aristotle-metadata-registry/aristotle_mdr/forms/search.py]: This is probably more descriptive.

File: python/aristotle-metadata-registry/aristotle_mdr/forms/search.py [hunk 2]
```diff
@@ -335,10 +348,11 @@ def search(self):
         if self.load_all:
             sqs = sqs.load_all()
 
-        # Only show models that are in apps that are enabled
+        # Populate the app lables that you can filter on
         app_labels = fetch_metadata_apps()
         app_labels.append('aristotle_mdr_help')
-        sqs = sqs.filter(django_ct_app_label__in=app_labels)
+        # TODO: Do we even need this?
+        # sqs = sqs.filter(django_ct_app_label__in=app_labels)
```
Comment (LegoStormtroopr) [inline python/aristotle-metadata-registry/aristotle_mdr/forms/search.py]: Yes, we do need this. Its so if a user has "disabled" a metadata module those wont show in results. I'll fix in my branch when I pull this in.

File: python/aristotle-metadata-registry/aristotle_mdr/search_indexes.py [hunk 3]
```diff
@@ -37,59 +38,32 @@ def prepare_template(self, obj):
...
-    text = ConceptFallbackCharField(document=True, use_template=True)
+class BaseObjectIndex(indexes.SearchIndex):
+    text = indexes.CharField(document=True, use_template=True)
     modified = indexes.DateTimeField(model_attr='modified')
     created = indexes.DateTimeField(model_attr='created')
     name = indexes.CharField(model_attr='name', boost=1)
     # Thanks ElasticSearch - https://github.com/django-haystack/django-haystack/issues/569
-    name_sortable = indexes.CharField(model_attr='name', indexed=False, stored=True)
-    django_ct_app_label = indexes.CharField()
+    # name_sortable = indexes.CharField(model_attr='name', indexed=False, stored=True)
```
Comment (LegoStormtroopr) [inline python/aristotle-metadata-registry/aristotle_mdr/search_indexes.py]: Need this back, will fix after pull

File: python/aristotle-metadata-registry/aristotle_mdr/templates/search/indexes/aristotle_mdr/discussionpost_text.txt [hunk 4]
```diff
@@ -0,0 +1,2 @@
+{{ object.title }}
+{{ object.body }}
```
Comment (LegoStormtroopr) [inline python/aristotle-metadata-registry/aristotle_mdr/templates/search/indexes/aristotle_mdr/discussionpost_text.txt]: Once this PR is in can you quickly add another PR that updates this template so it includes all the comments in the search text as well.

File: python/aristotle-metadata-registry/aristotle_mdr/templates/search/indexes/aristotle_mdr_stewards/collection_text.txt [hunk 5]
```diff
@@ -0,0 +1,2 @@
+{{ object.name }}
+{{ object.description }}
```
Comment (LegoStormtroopr) [inline python/aristotle-metadata-registry/aristotle_mdr/templates/search/indexes/aristotle_mdr_stewards/collection_text.txt]: In the same PR as above can your remove whichever of these txt templates isn't needed?

File: python/aristotle-metadata-registry/aristotle_mdr/utils/utils.py [hunk 6]
```diff
@@ -252,6 +252,7 @@ def fetch_metadata_apps():
     """
     aristotle_apps = list(fetch_aristotle_settings().get('CONTENT_EXTENSIONS', []))
     aristotle_apps += ["aristotle_mdr"]
+    aristotle_apps += ["aristotle_mdr_stewards"]
```
Comment (LegoStormtroopr) [inline python/aristotle-metadata-registry/aristotle_mdr/utils/utils.py]: I dont think theres any metadata content in this module.
Why is this being included in this list?

Comment (s-i-l-k-e) [reply python/aristotle-metadata-registry/aristotle_mdr/utils/utils.py]: This is to include Collections in the search

File: python/aristotle-metadata-registry/aristotle_mdr/signals.py [hunk 7]
```diff
@@ -57,6 +59,10 @@ def handle_concept_save(self, sender, instance, **kwargs):
             else:
                 return
 
+        from aristotle_mdr.models import DiscussionPost
```
[the remark this answers is not in the document]
Comment (s-i-l-k-e) [reply python/aristotle-metadata-registry/aristotle_mdr/signals.py]: This doesn't work, unfortunately

File: python/aristotle-metadata-registry/aristotle_mdr/templates/search/indexes/aristotle_mdr/discussionpost_text.txt (hunk 4 above)
Comment (s-i-l-k-e) [reply python/aristotle-metadata-registry/aristotle_mdr/templates/search/indexes/aristotle_mdr/discussionpost_text.txt]: Resolved

File: python/aristotle-metadata-registry/aristotle_mdr/templates/search/indexes/aristotle_mdr_stewards/collection_text.txt (hunk 5 above)
Comment (s-i-l-k-e) [reply python/aristotle-metadata-registry/aristotle_mdr/templates/search/indexes/aristotle_mdr_stewards/collection_text.txt]: Resolved

File: python/aristotle-metadata-registry/aristotle_mdr/utils/utils.py (hunk 6 above)
Comment (s-i-l-k-e) [reply python/aristotle-metadata-registry/aristotle_mdr/utils/utils.py]: Resolved in later pull request

**Pull request merged**